src/hb-subset-plan.cc

```diff
@@ -768,8 +853,10 @@ _populate_unicodes_to_retain (const hb_set_t *unicodes,
   }
 
   auto &arr = plan->unicode_to_new_gid_list;
+
   if (arr.length)
   {
+    arr.qsort (); // Decompositions are added in non-sorted order
```

Comment:
Since this is only needed if decompositions were added, maybe we can have the _fill_unicode_and_glyph_map return a bool to indicate if decompositions were added and only sort here when needed.

Also in some cases the list get's qsort'd prior to getting here (eg. line 825)